.github/workflows/test.yml

```diff
@@ -132,88 +122,113 @@ jobs:
         run: cat .github/env >> $GITHUB_ENV
 
       - name: Install stable toolchain
-        run: |
-          rustup install $RUST_STABLE_VERSION
-          rustup default $RUST_STABLE_VERSION
-          rustup target add wasm32-unknown-unknown
-          rustup component add rust-src
+        uses: dtolnay/rust-toolchain@master
+        with:
+          targets: "wasm32-unknown-unknown"
+          components: "rust-src"
+          toolchain: "${{env.RUST_STABLE_VERSION}}"
 
       - name: Fetch cache
-        uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43 # v2.7.0
+        uses: Swatinem/rust-cache@f0deed1e0edfc6a9be95417288c0e1099b1eeec3 # v2.7.7
         with:
-          shared-key: "fellowship-cache-tests"
+          shared-key: "fellowship-cache-integration-tests"
+          save-if: ${{ github.ref == 'refs/heads/main' }}
 
       - name: Test ${{ matrix.itest.name }}
         run: cargo test -p ${{ matrix.itest.package }} --release --locked -q
         env:
-          RUSTFLAGS: "-C debug-assertions" # FAIL-CI "-D warnings" re-enable after AHM dev branch
+          RUSTFLAGS: "-C debug-assertions -D warnings"
 
-  # Job required by "confirmTestPassed"
-  build-chain-spec-generator:
-    runs-on: ubuntu-latest
+  ahm-test:
+    runs-on: ubuntu-22.04
     steps:
       - name: Cancel previous runs
         uses: styfle/cancel-workflow-action@b173b6ec0100793626c2d9e6b90435061f4fc3e5 # v0.11.0
         with:
           access_token: ${{ github.token }}
 
-      - name: Install updates and protobuf-compiler
-        run: sudo apt update && sudo apt install --assume-yes cmake protobuf-compiler
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install updates and dependencies
+        run: .github/install-deps.sh
 
-      - name: Free space on the runner
+      - name: Set rust version via common env file
+        run: cat .github/env >> $GITHUB_ENV
+
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          targets: "wasm32-unknown-unknown"
+          components: "rust-src"
+          toolchain: "${{env.RUST_STABLE_VERSION}}"
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@f0deed1e0edfc6a9be95417288c0e1099b1eeec3 # v2.7.7
+        with:
+          shared-key: "fellowship-cache-integration-tests"
+          save-if: ${{ github.ref == 'refs/heads/main' }}
+
+      - name: Download AHM test snapshots
         run: |
-          df -h
-          sudo apt -y autoremove --purge
-          sudo apt -y autoclean
-          sudo rm -rf /usr/share/dotnet
-          sudo rm -rf /opt/ghc
-          sudo rm -rf "/usr/local/share/boost"
-          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
-          df -h
+          curl -sL 'https://sole-amaranth-pinniped.myfilebase.com/ipfs/QmTdrtye7Vt2xDybZHR3Xjf8kbxXkSfk54AUz2DuKKGmrG?download=true&filename=try-runtime-snap-polkadot-25075273.snap.lz4' -o polkadot.snap.lz4
```

Comment:
may be some general name so that we can just replace the file by newer version?

Reply:
This is far from ideal - we need some S3 bucket or something that we can update with a single command and that also replaces it here.  
I added this just as a first version, we can improve it then.